src/formatDate.js

```diff
@@ -49,8 +49,68 @@
  * @returns {string}
  */
 
+function getNewFormatYear(fromFormYear, toFormYear, oldFormatYear) {
+  let newFormatYear = oldFormatYear;
+
+  if (fromFormYear.length > toFormYear.length) {
+    newFormatYear = oldFormatYear.slice(2);
+  }
+
+  if (fromFormYear.length < toFormYear.length) {
+    newFormatYear = oldFormatYear < 30
+      ? 20 + '' + oldFormatYear
+      : 19 + '' + oldFormatYear;
+  }
+
+  return newFormatYear;
+}
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateInNewFormat = [];
+  const formatDates = date.split(fromFormat[3]);
+
+  const yearIndexFromFormat = fromFormat.indexOf('YY') >= 0
+    ? fromFormat.indexOf('YY')
+    : fromFormat.indexOf('YYYY');
```

Comment:
You can make the search easier by using `Math`:
```suggestion
   const yearIndexFromFormat = Math.max(fromFormat.indexOf('YY'), fromFormat.indexOf('YYYY'));
```
The same for `yearIndexToFormat`.